File: app/src/main/java/de/bikebean/app/ui/main/status/menu/log/GithubGistBody.java
package de.bikebean.app.ui.main.status.menu.log;

import androidx.annotation.NonNull;

import com.google.gson.Gson;

class GithubGistBody {

    private final @NonNull String description;
    private final @NonNull GithubGistFiles files;

    GithubGistBody(final @NonNull String description, final @NonNull String smsTsv,
                   final @NonNull String stateTsv, final @NonNull String logTsv) {
        this.description = description;
        this.files = new GithubGistFiles(smsTsv, stateTsv, logTsv);
    }

    @NonNull String createJsonApiBody() {
        final @NonNull Gson gson = new Gson();

        final @NonNull String json = gson.toJson(this);

        return json
                .replace("sms_tsv", "sms.tsv")
                .replace("state_tsv", "state.tsv")
                .replace("log_tsv", "log.tsv");
    }
}

File: app/src/main/java/de/bikebean/app/ui/drawer/log/GithubGistBody.kt
package de.bikebean.app.ui.drawer.log

import kotlinx.serialization.*
import kotlinx.serialization.json.*

@Serializable
data class GithubGistBody(
        val description: String,
        val files: GithubGistFiles,
        val public: Boolean)

internal class GithubGistBodyBuilder(
        description: String, smsTsv: String,
        stateTsv: String, logTsv: String) {

    private val json = Json.encodeToString(
            GithubGistBody(
                    description,
                    GithubGistFilesBuilder(smsTsv, stateTsv, logTsv).build(),
                    false
            )
    )

    fun build(): String {
        return json
                .replace("sms_tsv", "sms.tsv")
                .replace("state_tsv", "state.tsv")
                .replace("log_tsv", "log.tsv")
    }

}